internal/mode/static/manager.go

```diff
@@ -125,6 +125,20 @@ func StartManager(cfg config.Config) error {
 		return fmt.Errorf("cannot clear NGINX configuration folders: %w", err)
 	}
 
+	// Ensure NGINX is running before registering metrics & starting the manager.
+	if err := ngxruntime.EnsureNginxRunning(ctx); err != nil {
+		return fmt.Errorf("NGINX is not running: %w", err)
+	}
+
+	var mgrCollector ngxruntime.ManagerCollector
+	mgrCollector = nkgmetrics.NewManagerFakeCollector()
+	if cfg.MetricsConfig.Enabled {
+		mgrCollector, err = configureNginxMetrics(cfg.GatewayClassName)
```

Comment:
I think `configureNginxMetrics` became a potential source of confusion -- the reader will need to read it definition to really understand what's going on

the way it is invoked, it looks like it's just responsible for creating the mgrCollector. However, in fact, it does many things - creates collectors, tries to register them, and then returns one of them.

I suggest getting rid of it. I think replacing it with something like below will make it clearer:
```go
if cfg.MetricsConfig.Enabled {
		constLabels := map[string]string{"class": cfg.GatewayClassName}

		ngxCollector, err := nkgmetrics.NewNginxMetricsCollector(constLabels)
		if err != nil {
			return fmt.Errorf("cannot get NGINX metrics: %w", err)
		}
		mgrCollector = nkgmetrics.NewManagerMetricsCollector(constLabels)

		if err := registerCollectors(mgrCollector, ngxCollector); err != nil {
			return err
		}
	}
```

where registerCollectors is a new helper function that register collectors.

Reply:
Hmm, yes it is a good point that it is confusing naming, but the `StartManager` function is already getting pretty complex so I do like having a separate helper function for the metrics collectors. I've refactored it a bit and added more comments, what do you think?